src/blocks/menu-item/edit.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * Internal dependencies.
+ */
+import { hasEmptyAttributes } from '../../utils/block-helpers';
+
+/**
+ * WordPress dependencies.
+ */
+const { __ } = wp.i18n;
+const { Component, Fragment } = wp.element;
+const { PanelBody, ToggleControl, IconButton, Toolbar } = wp.components;
+const { dispatch, select } = wp.data;
+const {
+	InspectorControls,
+	RichText,
+	MediaPlaceholder,
+	MediaUpload,
+	BlockControls,
+} = wp.editor;
+
+/**
+ * Handle creation and removal of placeholder elements so that we always have one available to use.
+ *
+ * @param {Integer} childClientId The child block's ClientId.
+ */
+const handlePlaceholderPlacement = childClientId => {
+	const menuClientId = select( 'core/editor' ).getBlockRootClientId(
+		childClientId
+	);
+
+	const menuItems = select( 'core/editor' ).getBlocksByClientId( menuClientId )[ 0 ]
+		.innerBlocks;
+
+	const placeholders = menuItems.filter(
+		item => item.name === 'coblocks/menu-item' && isEmpty( item.attributes )
+	);
+
+	// Add a placeholder if there are none. Remove trailing placholders if there are more than one.
+	if ( placeholders.length === 0 ) {
+		const newMenuItem = wp.blocks.createBlock( 'coblocks/menu-item', {} );
+		dispatch( 'core/editor' ).insertBlocks(
+			newMenuItem,
+			menuItems.length,
+			menuClientId,
+			false
+		);
+	} else if ( placeholders.length > 1 ) {
```

Comment:
@jrtashjian Normally when I see an else statement I think there might be a better way. What about a method that tells you explicitely what you are about to do with only 1 if that returns early. Maybe something like `dispatchCoreInsertBlock` or something.

Reply:
Good point. I came up with a new solution that avoids the else statement. Decided to just always remove all placeholder blocks categorized as trailing when this runs. We now only check the length to add a placeholder block.

```
dispatch( 'core/editor' ).removeBlocks(
    placeholders
        .filter( ( item, index ) => item.clientId !== childClientId && index !== 0 )
        .map( item => item.clientId ),
    false
);
```